src/text/line.rs

```diff
@@ -10,35 +10,37 @@ use crate::prelude::*;
 /// text. When a [`Line`] is rendered, it is rendered as a single line of text, with each [`Span`]
 /// being rendered in order (left to right).
 ///
-/// [`Line`]s can be created from [`Span`]s, [`String`]s, and [`&str`]s. They can be styled with a
-/// [`Style`], and have an [`Alignment`].
-///
-/// The line's [`Alignment`] is used by the rendering widget to determine how to align the line
-/// within the available space. If the line is longer than the available space, the alignment is
-/// ignored and the line is truncated.
-///
-/// The line's [`Style`] is used by the rendering widget to determine how to style the line. If the
-/// line is longer than the available space, the style is applied to the entire line, and the line
-/// is truncated. Each [`Span`] in the line will be styled with the [`Style`] of the line, and then
-/// with its own [`Style`].
-///
-/// `Line` implements the [`Widget`] trait, which means it can be rendered to a [`Buffer`]. Usually
-/// apps will use the [`Paragraph`] widget instead of rendering a [`Line`] directly as it provides
-/// more functionality.
-///
 /// # Constructor Methods
 ///
 /// - [`Line::default`] creates a line with empty content and the default style.
 /// - [`Line::raw`] creates a line with the given content and the default style.
 /// - [`Line::styled`] creates a line with the given content and style.
 ///
+/// # Conversion Methods
+///
+/// - [`Line::from`] creates a `Line` from a [`String`].
+/// - [`Line::from`] creates a `Line` from a [`&str`].
+/// - [`Line::from`] creates a `Line` from a [`Vec`] of [`Span`]s.
+/// - [`Line::from`] creates a `Line` from single [`Span`].
+/// - [`String::from`] converts a line into a [`String`].
+/// - [`Line::from_iter`] creates a line from an iterator of items that are convertible to [`Span`].
+///
 /// # Setter Methods
 ///
 /// These methods are fluent setters. They return a `Line` with the property set.
 ///
 /// - [`Line::spans`] sets the content of the line.
 /// - [`Line::style`] sets the style of the line.
 /// - [`Line::alignment`] sets the alignment of the line.
+/// - [`Line::left_aligned`] sets the alignment of the line to [`Alignment::Left`].
+/// - [`Line::centered`] sets the alignment of the line to [`Alignment::Center`].
+/// - [`Line::right_aligned`] sets the alignment of the line to [`Alignment::Right`].
```

Comment:
Is it really worth it to have these in the doc comment? `cargo doc` / <docs.rs> lists them relatively obviously and will always be up-to-date as it's generated from the source. This is basically duplicated documentation.

Conversions and iteration methods could be displayed more nicely by the docs, so I see the point of including them.

Reply:
The idea / rationale behind this is to provide a list of methods upfront that helps newer rust users find what they need, and groups the methods in a way that makes it easier to understand the general ideas of the types. This is more difficult to see from the auto generated summary.

Reply:
It's added work that I don't like, but I must say I usually have a hard time finding what I'm looking for in the long list of methods (in ratatui or other libs) without the text search (which suppose I know what I'm looking for).
Plus we can list other methods not associated with the type like here https://github.com/ratatui-org/ratatui/pull/969/files#diff-d8aa1f145d6f63ab98ad22cb5d47fc3cd79b4a4fb3b7f0c407c3eecc5c19dafdR25